include/forcing/ForcingsEngineDataProvider.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef NGEN_FORCING_FE_PER_FEATURE_DATA_PROVDER_HPP
+#define NGEN_FORCING_FE_PER_FEATURE_DATA_PROVDER_HPP
+
+#include "GenericDataProvider.hpp"
+#include "ForcingsEngine.hpp"
+
+namespace data_access {
+
+struct ForcingsEngineDataProvider
```

Comment:
Having just gone over the `DataProvider` class hierarchy, I think what we want here may be to declare
```suggestion
template <class data_type, class selection_type>
struct ForcingsEngineDataProvider : public DataProvider<data_type, selection_type>
```

And then have further derived classes `ForcingsEngineLumpedDataProvider` (in this PR), `ForcingsEngineGridDataProvider`, and `ForcingsEngineMeshDataProvider` (each to come later).

Along the way, we could do away with `GenericDataProvider` as a separate class definition, and just make it an alias
`using GenericDataProvider = DataProvider<double, CatchmentAggrDataSelector>;`
So that `ForcingsEngineLumptedDataProvider` would be a super-type and hence pointers would be convertible.

All of the above implies that `GenericDataProvider` should probably be renamed to something like `CatchmentLumpedDataprovider`, if it's kept at all.